hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieHiveCatalog.java

```diff
@@ -117,21 +117,24 @@ public class HoodieHiveCatalog extends AbstractCatalog {
 
   // optional catalog base path: used for db/table path inference.
   private final String catalogPath;
+  private final boolean external;
 
-  public HoodieHiveCatalog(String catalogName, String catalogPath, String defaultDatabase, String hiveConfDir) {
-    this(catalogName, catalogPath, defaultDatabase, HoodieCatalogUtil.createHiveConf(hiveConfDir), false);
+  public HoodieHiveCatalog(String catalogName, String catalogPath, String defaultDatabase, String hiveConfDir, boolean external) {
+    this(catalogName, catalogPath, defaultDatabase, HoodieCatalogUtil.createHiveConf(hiveConfDir), false, external);
   }
 
   public HoodieHiveCatalog(
       String catalogName,
       String catalogPath,
       String defaultDatabase,
       HiveConf hiveConf,
-      boolean allowEmbedded) {
+      boolean allowEmbedded,
+      boolean external) {
```

Comment:
Nice catch, can we put the param `external` in front of `allowEmbedded` and can we add a test case in `TestHoodieHiveCatalog`.